### PR TITLE
fix: pool timeout bug

### LINKED ISSUE #37: Pool timeout error
There appears to be an issue with the DB that happens roughly 20-30 minutes after the Radio is started/restarted. So far we've only noticed this happening in the Radio running in our kubernetes cluster, when the Radio is ran locally it works fine. Examples of the error:
1. In the GraphQL playground:
```graphql
{
  "data": null,
  "errors": [
    {
      "message": "pool timed out while waiting for an open connection",
      "locations": [
        {
          "line": 2,
          "column": 3
        }
      ],
      "path": [
        "queryActiveIndexers"
      ]
    }
  ]
}
```

2. In the logs:
```bash
  2024-03-12T09:51:33.917273Z  WARN listener_radio::operator: Error during pruning by retention, err: pool timed out while waiting for an open connection
    at src/operator/mod.rs:198

  2024-03-12T09:51:36.918274Z  INFO listener_radio::operator: Monitoring summary, total_messages: 0, total_num_pruned: 0
 ```
 
 This seems to be related to the message pruning logic.

### PR DESCRIPTION
**Pull request by cargopete**
- [x] added `count_messages` resolver
- [x] made `prune_old_messages` more efficient
- [x] renamed `comparison_interval` to `summary_interval` to match what the function is doing a bit better, also increased the interval to 180 seconds
- [x] using `count_messages` instead of `list_messages` in summary function
- [x] other small changes

### CODE REVIEW

**Review by hopeyen — DISMISSED** [review]
:ship:

**Comment by cargopete** [comment]
switching to draft until I fix these damn CI issues

File: src/operator/mod.rs [hunk 1]
```diff
@@ -185,10 +185,12 @@ impl RadioOperator {
                         };
                     }
 
+                    let batch_size = 1000;
```
Comment (hopeyen) [inline src/operator/mod.rs:188]: how was this size determined? would like to understand if this is a magic number or if you had some analysis on the optimal size; either way would appreciate a comment in code

**Review by hopeyen — APPROVED** [review]
damn the ci

File: src/operator/mod.rs (hunk 1 above)
Comment (cargopete) [reply src/operator/mod.rs:188]: magic number really :> i take recommendations but it seemed like an ok batch size

**Pull request merged**